pkg/asset/installconfig/aws/validation.go

```diff
@@ -47,18 +48,32 @@ func Validate(ctx context.Context, meta *Metadata, config *types.InstallConfig)
 	allErrs = append(allErrs, validateAMI(ctx, config)...)
 	allErrs = append(allErrs, validatePlatform(ctx, meta, field.NewPath("platform", "aws"), config.Platform.AWS, config.Networking, config.Publish)...)
 
-	if config.ControlPlane != nil && config.ControlPlane.Platform.AWS != nil {
-		allErrs = append(allErrs, validateMachinePool(ctx, meta, field.NewPath("controlPlane", "platform", "aws"), config.Platform.AWS, config.ControlPlane.Platform.AWS, controlPlaneReq)...)
+	if config.ControlPlane != nil {
+		allErrs = append(allErrs, validateArchitecture(ctx, field.NewPath("controlPlane", "architecture"), config.ControlPlane.Architecture)...)
```

Comment:
This is not the correct place for this validation. Static validation should be in `types/aws/validation`.

Reply:
fixed and added as a separate commit: https://github.com/openshift/installer/pull/4870/commits/03d6b77734f452aa9a8ab76533a3c207c5715913